financialaid/serializers.py

```diff
@@ -61,3 +83,77 @@ def save(self):
+        ],
+        write_only=True
+    )
+    tier_program_id = IntegerField(write_only=True)
+
+    def validate(self, data):
+        """
+        Validators for this serializer
+        """
+        # Check that the previous financial aid status allows for the new status
+        if (data['action'] == FinancialAidStatus.REJECTED and
+                self.instance.status != FinancialAidStatus.PENDING_MANUAL_APPROVAL):
+            raise ValidationError("Cannot reject application that is not pending manual approval.")
+        if (data['action'] == FinancialAidStatus.APPROVED and
+                self.instance.status != FinancialAidStatus.PENDING_MANUAL_APPROVAL):
+            raise ValidationError("Cannot approve application that is not pending manual approval.")
+        if (data['action'] == FinancialAidStatus.PENDING_MANUAL_APPROVAL and
+                self.instance.status != FinancialAidStatus.PENDING_DOCS):
+            raise ValidationError("Cannot mark documents as received for application not pending docs.")
+        # Check tier program exists
+        try:
+            data["tier_program"] = TierProgram.objects.get(
+                id=data["tier_program_id"],
+                program_id=self.instance.tier_program.program_id,
+                current=True
+            )
+        except TierProgram.DoesNotExist:
+            raise ValidationError("Financial Aid Tier does not exist for this program.")
+        return data
+
+    def save(self):
+        """
+        Save method for this serializer
+        """
+        tier_program = self.validated_data["tier_program"]
+        self.instance.status = self.validated_data["action"]
+        if self.instance.status == FinancialAidStatus.APPROVED:
+            self.instance.tier_program = tier_program
+            email_data = {
+                "email_subject": FINANCIAL_AID_APPROVAL_SUBJECT_TEXT,
+                "email_body": FINANCIAL_AID_APPROVAL_MESSAGE_BODY,
+                "email_recipient": self.instance.user.email
+            }
+        elif self.instance.status == FinancialAidStatus.REJECTED:
+            self.instance.tier_program = get_no_discount_tier_program(self.instance.tier_program.program_id)
+            email_data = {
+                "email_subject": FINANCIAL_AID_REJECTION_SUBJECT_TEXT,
+                "email_body": FINANCIAL_AID_REJECTION_MESSAGE_BODY,
+                "email_recipient": self.instance.user.email
+            }
+        elif self.instance.status == FinancialAidStatus.PENDING_MANUAL_APPROVAL:
+            email_data = {
+                "email_subject": FINANCIAL_AID_DOCUMENTS_SUBJECT_TEXT,
+                "email_body": FINANCIAL_AID_DOCUMENTS_MESSAGE_BODY,
+                "email_recipient": self.instance.user.email
+            }
+        else:
+            raise Exception("Error in ")
+        self.instance.save()
+        # Send email notification
+        FinancialAidMailView().post(self.context["request"], **email_data)
```

Comment:
Referenced in https://github.com/mitodl/micromasters/pull/1096#discussion_r80290727